.dotnet/src/Generated/Models/MessageObjectIncompleteDetails.Serialization.cs

```diff
@@ -0,0 +1,130 @@
+// <auto-generated/>
+
+using System;
+using OpenAI.ClientShared.Internal;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text.Json;
+using OpenAI;
+
+namespace OpenAI.Internal.Models
+{
+    internal partial class MessageObjectIncompleteDetails : IJsonModel<MessageObjectIncompleteDetails>
+    {
+        void IJsonModel<MessageObjectIncompleteDetails>.Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<MessageObjectIncompleteDetails>)this).GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(MessageObjectIncompleteDetails)} does not support '{format}' format.");
+            }
+
+            writer.WriteStartObject();
+            writer.WritePropertyName("reason"u8);
+            writer.WriteStringValue(Reason);
+            if (options.Format != "W" && _serializedAdditionalRawData != null)
+            {
+                foreach (var item in _serializedAdditionalRawData)
+                {
+                    writer.WritePropertyName(item.Key);
+#if NET6_0_OR_GREATER
```

Comment:
this might be a common pattern. Should we add a helper for writing raw values somewhere and then just use this helper.

Reply:
100% agreed; there's a lot more here in the custom code folder than we'd like, but this (along with several other patterns) would be great candidates for codegen-level refactor -- even if just for readability.

Reply:
@annelo-msft, do we have a convention for how to organize such polyfills in our sources?

Reply:
The short answer is that we do in Core and ClientModel, but in client code we're either using shared source or the generator is writing the `#if` sequence directly.

I agree that it would be nice to have a better answer - let me flag this as work to consider as part of https://github.com/Azure/azure-sdk-for-net/issues/28369.